### PR TITLE
CIRC-1430

### PR DESCRIPTION
**Pull request by Mykyta-Varenyk**
Resolves: https://issues.folio.org/browse/CIRC-1430

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/MoveRequestService.java [hunk 1]
```diff
@@ -78,6 +80,12 @@ private RequestAndRelatedRecords pagedRequestIfDestinationItemAvailable(
     return requestAndRelatedRecords;
   }
 
+  private CompletableFuture<Result<RequestAndRelatedRecords>> validateItemInstanceId(
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/MoveRequestService.java:83]: Why do we compose it and wrap the result in completedFuture? Doesn't it make more sense to use thenApply and map 'records' with refuseWhenItemToBeMovedIsFromDifferentInstance directly?

File: src/main/java/org/folio/circulation/domain/RequestServiceUtility.java [hunk 2]
```diff
@@ -115,6 +115,23 @@ private RequestServiceUtility() { }
     }
   }
 
+  static Result<RequestAndRelatedRecords> refuseWhenItemToBeMovedIsFromDifferentInstance(
+    RequestAndRelatedRecords requestAndRelatedRecords) {
+
+    Request request = requestAndRelatedRecords.getRequest();
+    Item item = request.getItem();
+    if (!Objects.equals(item.getInstanceId(), request.getInstanceId())) {
+      String message = "Request can be moved only to an item with the same instance ID";
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/RequestServiceUtility.java:124]: If we're using this value only once, does it make sense to make it a variable?

Comment (Mykyta-Varenyk) [reply src/main/java/org/folio/circulation/domain/RequestServiceUtility.java]: Refactored

File: src/test/java/api/requests/scenarios/MoveRequestTests.java [hunk 3]
```diff
@@ -7,9 +7,11 @@
 import static api.support.matchers.EventMatchers.isValidLoanDueDateChangedEvent;
 import static api.support.matchers.ItemStatusCodeMatcher.hasItemStatus;
 import static api.support.matchers.TextDateTimeMatcher.isEquivalentTo;
+import static api.support.matchers.ValidationErrorMatchers.*;
```
Comment (viktor-draban) [inline src/test/java/api/requests/scenarios/MoveRequestTests.java]: remove '*'

File: src/test/java/api/requests/scenarios/MoveRequestTests.java [hunk 4]
```diff
@@ -7,9 +7,11 @@
...
 import static java.time.Clock.fixed;
 import static java.time.Clock.offset;
 import static java.time.Duration.ofDays;
+import static java.util.function.Function.*;
```
Comment (viktor-draban) [inline src/test/java/api/requests/scenarios/MoveRequestTests.java]: remove '*'

File: src/test/java/api/requests/scenarios/MoveRequestTests.java (hunk 3 above)
Comment (Mykyta-Varenyk) [reply src/test/java/api/requests/scenarios/MoveRequestTests.java]: Done

File: src/test/java/api/requests/scenarios/MoveRequestTests.java (hunk 4 above)
Comment (Mykyta-Varenyk) [reply src/test/java/api/requests/scenarios/MoveRequestTests.java]: Done

**Review by viktor-draban — APPROVED** [review]

File: src/main/java/org/folio/circulation/domain/RequestServiceUtility.java [hunk 5]
```diff
@@ -115,6 +115,24 @@ private RequestServiceUtility() { }
     }
   }
 
+  static Result<RequestAndRelatedRecords> refuseWhenItemToBeMovedIsFromDifferentInstance(
```
Comment (Mykyta-Varenyk) [inline src/main/java/org/folio/circulation/domain/RequestServiceUtility.java:118]: Rename

File: src/main/java/org/folio/circulation/domain/RequestServiceUtility.java [hunk 6]
```diff
@@ -115,6 +115,24 @@ private RequestServiceUtility() { }
...
+    RequestAndRelatedRecords requestAndRelatedRecords) {
+
+    Request request = requestAndRelatedRecords.getRequest();
+    Item item = request.getItem();
+    if (!Objects.equals(item.getInstanceId(), request.getInstanceId())) {
+      HashMap<String, String> parameters = new HashMap<>();
+      parameters.put(ITEM_ID, request.getItemId());
+      parameters.put("originalInstanceId", request.getInstanceId());
```
Comment (Mykyta-Varenyk) [inline src/main/java/org/folio/circulation/domain/RequestServiceUtility.java:126]: pass only new instance id

File: src/test/java/api/support/fixtures/ItemsFixture.java [hunk 7]
```diff
@@ -331,6 +341,19 @@ public InstanceBuilder instanceBasedUponSmallAngryPlanet() {
       getPersonalContributorNameTypeId());
   }
 
+  public List<ItemResource> createWithCallNumberStringComponents(String... components) {
```
Comment (Mykyta-Varenyk) [inline src/test/java/api/support/fixtures/ItemsFixture.java:344]: Refactor

File: src/main/java/org/folio/circulation/domain/RequestServiceUtility.java (hunk 6 above)
Comment (Mykyta-Varenyk) [reply src/main/java/org/folio/circulation/domain/RequestServiceUtility.java]: Fixed

File: src/main/java/org/folio/circulation/domain/RequestServiceUtility.java (hunk 5 above)
Comment (Mykyta-Varenyk) [reply src/main/java/org/folio/circulation/domain/RequestServiceUtility.java]: Renamed

File: src/test/java/api/support/fixtures/ItemsFixture.java [hunk 8]
```diff
@@ -332,15 +344,22 @@ public InstanceBuilder instanceBasedUponSmallAngryPlanet() {
   }
 
   public List<ItemResource> createMultipleItemsForTheSameInstance(int size) {
+    return createMultipleItemForTheSameInstanceWithItemAdditionalProperties(
+      new ArrayList<>(Collections.nCopies(size, identity())));
+  }
+
+  public List<ItemResource> createMultipleItemForTheSameInstanceWithItemAdditionalProperties(
```
Comment (Mykyta-Varenyk) [inline src/test/java/api/support/fixtures/ItemsFixture.java:351]: rename

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**